### DISCUSSION TITLE
Not able to load graphs on kiali.

### DISCUSSION THREAD

**Discussion by bj-1795** [kiali/kiali#4539] [Q&A]

**Comment by bj-1795** [comment]
I have installed kiali via operator .

KialiCR.yml
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  name: kiali
  annotations:
    ansible.sdk.operatorframework.io/verbosity: "1"
spec:
  istio_namespace: istio-system
  version: "default"
  namespaces:
    exclude:
      - "istio-operator"
      - "kube-.*"
      - "openshift.*"
      - "ibm.*"
  auth:
    strategy: "anonymous"
  deployment:
    accessible_namespaces : "**"
    hpa:
      spec:
        maxReplicas: 4
        minReplicas: 2
        targetCPUUtilizationPercentage: 80
    ingress:
      enabled: true
      class_name: "aws-load-balancer"
      override_yaml:
        metadata:
          annotations:
            alb.ingress.kubernetes.io/backend-protocol: HTTP
            alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:ap-southeast-1:917596758507:certificate/ce26aef4-5fe6-419b-9456-9d20101fbee9
            alb.ingress.kubernetes.io/group.name: scb-kiali-poc-internal-ingress
            alb.ingress.kubernetes.io/healthcheck-protocol: HTTP
            alb.ingress.kubernetes.io/inbound-cidrs: 10.0.0.0/8,192.168.0.0/16
            alb.ingress.kubernetes.io/ip-address-type: ipv4
            alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS": 443}]'
            alb.ingress.kubernetes.io/success-codes: '200,404,302'
            alb.ingress.kubernetes.io/load-balancer-attributes: access_logs.s3.enabled=false
            alb.ingress.kubernetes.io/scheme: internal
            alb.ingress.kubernetes.io/ssl-policy: ELBSecurityPolicy-2016-08
            alb.ingress.kubernetes.io/subnets: subnet-07cbe15ecd89a4ae5,subnet-002dbd98d22fce42c,subnet-054c2e9ff2ba38075
            alb.ingress.kubernetes.io/tags: Company=aella,Project=kiali-poc,Environment=sandbox
            alb.ingress.kubernetes.io/target-type: ip
      spec:
        rules:
        - http:
            paths:
            - path: /kiali
              backend:
                serviceName: kiali
                servicePort: 20001
        
    requests:
      cpu: "1"
      memory: "2Gi"
    limits:
      cpu: "2"
      memory: "3Gi"
  server:
    web_fqdn: "sandbox-kiali.aella-play.net"
   
I am not able to get graphs on kiali dashboard .
Getting below error in logs :
horizontalpodautoscaler.autoscaling/kiali   Deployment/kiali   <unknown>/80%   2         4         2          44m
bjha@WKWIN5512954:/mnt/c/Users/bhajha/Documents$ kubectl logs -f kiali-5bbbc6ccff-5wd9b   -n istio-system
2021-12-02T15:00:10Z INF Kiali: Version: v1.43.0, Commit: be72259771ba9cd1e4058c496a566d1b4e28206b

2021-12-02T15:00:10Z INF Using authentication strategy [anonymous]
2021-12-02T15:00:10Z WRN Kiali auth strategy is configured for anonymous access - users will not be authenticated.
2021-12-02T15:00:10Z INF Kiali: Console version: 1.43.0
2021-12-02T15:00:10Z INF Server endpoint will start at [:20001/kiali]
2021-12-02T15:00:10Z INF Server endpoint will serve static content from [/opt/kiali/console]
2021-12-02T15:00:10Z INF Starting Metrics Server on [:9090]
2021-12-02T15:03:17Z INF Kiali Cache is active for namespaces [.*]
2021-12-02T15:03:17Z INF [Prom Cache] Enabled
2021-12-02T15:03:17Z INF Waiting for Kiali cache for [namespace: istio-system] to sync
2021-12-02T15:03:17Z ERR grafana URL is not set in Kiali configuration
E1202 15:03:17.560350       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/reflector.go:167: Failed to watch *v1alpha3.WorkloadGroup: failed to list *v1alpha3.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1202 15:03:18.705164       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/reflector.go:167: Failed to watch *v1alpha3.WorkloadGroup: failed to list *v1alpha3.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1202 15:03:20.660764       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/reflector.go:167: Failed to watch *v1alpha3.WorkloadGroup: failed to list *v1alpha3.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1202 15:03:24.514580       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/reflector.go:167: Failed to watch *v1alpha3.WorkloadGroup: failed to list *v1alpha3.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1202 15:03:33.234019       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/reflector.go:167: Failed to watch *v1alpha3.WorkloadGroup: failed to list *v1alpha3.WorkloadGroup: the server could not find the requested resource (get workloadgroups.networking.istio.io)

**Comment by lucasponce** [comment]
Can you add details of the Istio version, the platform/cluster and more details ?

The error in the log means that the workloadgroups api of Istio is not present and that could mean that you're trying to install Kiali in an old (non supported) Istio version.

**Reply by jmazzitelli** [reply]
> The error in the log means that the workloadgroups api of Istio is not present and that could mean that you're trying to install Kiali in an old (non supported) Istio version

This is the answer.

**Comment by jshaughn** [comment]
There are many reasons you may not see a graph, this is unlikely to be a bug in Kiali.  In the future please start with asking questions on [Slack](https://app.slack.com/client/T382U9E4U/CJVRXB09F), or as a github discussion (as opposed to an issue). 
First thing to check is whether you actually have metrics in Prometheus, see https://kiali.io/docs/faq/graph/#queryprom for how to check.

**Comment by bj-1795** [comment]
@jshaughn @lucasponce I am able to get metrics in prometheus.

istio version - 1.7.2 on AWS EKS

**Comment by jmazzitelli** [comment]
1.  Can you wrap the yaml in github triple-ticks so the yaml is formatted. Your description shows yaml but is not formatted so it is hard to read.
2. I'll convert this to a discussion - so far this does not warrant a bug report. If it is truly a bug, we can convert back to bug/issue.

**Comment by jmazzitelli** [comment]
> istio version - 1.7.2 on AWS EKS

Kiali 1.43 is not compatible with Istio 1.7.2 -- see compatibility chart: https://kiali.io/docs/installation/installation-guide/prerequisites/#istio-version-compatibility
